common/tools/dev-tool/src/util/samples/transforms.ts

```diff
@@ -176,3 +186,55 @@ function namedImportsToObjectBindingPattern(
     )
   );
 }
+
+/**
+ * Processes a segmented module path to return the first segment. This is useful for packages that have nested imports
+ * such as "dayjs/plugin/duration".
+ *
+ * @param specifier - the module specifier to resolve to a package name
+ * @returns a package name
+ */
+export function resolveModule(specifier: string): string {
+  const parts = specifier.split("/", 2);
+
+  // The first part could be a namespace, in which case we need to join them
+  if (parts.length > 1 && parts[0].startsWith("@")) return parts[0] + "/" + parts[1];
+  else return parts[0];
+}
+
+/**
+ * Determines if a module specifier refers to a node builtin.
+ *
+ * @param specifier - the module specifier to test
+ */
+export function isNodeBuiltin(moduleSpecifier: string): boolean {
+  return (
+    moduleSpecifier.startsWith("node:") || nodeBuiltins.includes(resolveModule(moduleSpecifier))
```

Comment:
Nice catch on the `node:` specifier 😄

Reply:
Yeah this is something @xirzec brought up a week or so ago with me, but we were never testing for it before. Now it'll be handled.